avogadro/io/cmlformat.cpp

```diff
@@ -533,13 +540,17 @@ bool CmlFormat::write(std::ostream& out, const Core::Molecule& mol)
       Elements::symbol(a.atomicNumber());
     if (cell) {
       Vector3 fracPos = cell->toFractional(a.position3d());
-      atomNode.append_attribute("xFract") = formatNumber(floatStream,fracPos.x()).c_str();
-      atomNode.append_attribute("yFract") = formatNumber(floatStream,fracPos.y()).c_str();
-      atomNode.append_attribute("zFract") = formatNumber(floatStream,fracPos.z()).c_str();
+      atomNode.append_attribute("xFract") = formatNumber(numberStream,fracPos.x()).c_str();
+      atomNode.append_attribute("yFract") = formatNumber(numberStream,fracPos.y()).c_str();
+      atomNode.append_attribute("zFract") = formatNumber(numberStream,fracPos.z()).c_str();
     } else {
-      atomNode.append_attribute("x3") = formatNumber(floatStream,a.position3d().x()).c_str();
-      atomNode.append_attribute("y3") = formatNumber(floatStream,a.position3d().y()).c_str();
-      atomNode.append_attribute("z3") = formatNumber(floatStream,a.position3d().z()).c_str();
+      atomNode.append_attribute("x3") = formatNumber(numberStream,a.position3d().x()).c_str();
+      atomNode.append_attribute("y3") = formatNumber(numberStream,a.position3d().y()).c_str();
+      atomNode.append_attribute("z3") = formatNumber(numberStream,a.position3d().z()).c_str();
+    }
+    if(a.formalCharge() != 0) {
+      atomNode.append_attribute("formalCharge") =
+          formatNumber(numberStream, int(a.formalCharge())).c_str();
```

Comment:
This shouldn't be needed, since it's an integer. We ran into problems with CML expecting the "C" locale and floats written as "1,0456"